src/main/java/org/venice/beachfront/bfapi/config/BfApiWebSecurity.java

```diff
@@ -0,0 +1,67 @@
+package org.venice.beachfront.bfapi.config;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.AuthenticationDetailsSource;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.venice.beachfront.bfapi.auth.ApiKeyAuthProvider;
+import org.venice.beachfront.bfapi.auth.ExtendedRequestDetails;
+import org.venice.beachfront.bfapi.auth.FailedAuthEntryPoint;
+
+/**
+ * Initializes the security providers for validating API Keys on all requests that require API Key validation.
+ */
+@EnableAutoConfiguration
+@ComponentScan
+@Configuration
+public class BfApiWebSecurity extends WebSecurityConfigurerAdapter {
+
+	@Autowired
+	private ApiKeyAuthProvider apiKeyAuthProvider;
+
+	@Autowired
+	private FailedAuthEntryPoint failureEntryPoint;
+
+	@Override
+	protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+		auth.authenticationProvider(apiKeyAuthProvider);
+	}
+
+	@Override
+	public void configure(WebSecurity web) throws Exception {
+		web.ignoring().antMatchers("/").antMatchers("/index").antMatchers("/login").antMatchers("/oauth/start").antMatchers(HttpMethod.OPTIONS);
+	}
+
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
+		// @formatter:off
```

Comment:
Wow, I had no idea this comment existed. This is awesome!

Reply:
Yeah that's pretty cool